papaya-web/src/main/java/pl/edu/agh/papaya/rest/projects/service/AvailabilityRestService.java

```diff
@@ -8,59 +8,40 @@
 import pl.edu.agh.papaya.api.model.AvailabilityDto;
 import pl.edu.agh.papaya.mappers.AvailabilityMapper;
 import pl.edu.agh.papaya.model.Availability;
-import pl.edu.agh.papaya.model.Project;
 import pl.edu.agh.papaya.model.Sprint;
 import pl.edu.agh.papaya.model.SprintState;
 import pl.edu.agh.papaya.security.UserContext;
 import pl.edu.agh.papaya.service.availability.AvailabilityService;
-import pl.edu.agh.papaya.service.sprint.SprintService;
 import pl.edu.agh.papaya.util.ForbiddenAccessException;
-import pl.edu.agh.papaya.util.ResourceNotFoundException;
 
 @Service
 @RequiredArgsConstructor
 public class AvailabilityRestService {
 
-    private final SprintService sprintService;
+    private final SprintsRestService sprintRestService;
 
     private final UserContext userContext;
 
     private final AvailabilityService availabilityService;
 
     private final AvailabilityMapper availabilityMapper;
 
-    private final ProjectsRestService projectsRestService;
-
     public ResponseEntity<AvailabilityDto> getUserAvailability(Long projectId, Long sprintId) {
-        Project project = projectsRestService.getValidProject(projectId);
-        Sprint sprint = getValidSprint(sprintId);
-
-        if (!isSprintInProject(sprint, project)) {
-            throw new ForbiddenAccessException();
-        }
+        Sprint sprint = sprintRestService.getValidSprint(projectId, sprintId);
 
         AvailabilityDto availability = availabilityService
-                .getBySprintIdAndUserId(sprintId, userContext.getUserIdn())
+                .getBySprintIdAndUserId(sprint.getId(), userContext.getUserIdn())
                 .map(availabilityMapper::mapToApi)
                 .orElseGet(() -> new AvailabilityDto().timeAvailable(0L).timeRemaining(0L).notes(""));
 
         return ResponseEntity.ok(availability);
     }
 
-    private Sprint getValidSprint(Long sprintId) {
-        return sprintService.getById(sprintId)
-                .orElseThrow(ResourceNotFoundException::new);
-    }
-
-    private boolean isSprintInProject(Sprint sprint, Project project) {
-        return sprint.getProject().getId().equals(project.getId());
-    }
-
     public ResponseEntity<AvailabilityDto> updateUserAvailability(AvailabilityDto availabilityDto,
             Long projectId, Long sprintId) {
-        Project project = projectsRestService.getValidProject(projectId);
-        Sprint sprint = getValidSprint(sprintId);
-        if (!isSprintInProject(sprint, project) || isAvailabilityDeclarationNotAllowed(sprint)) {
+        Sprint sprint = sprintRestService.getValidSprint(projectId, sprintId);
+
+        if (isAvailabilityDeclarationNotAllowed(sprint)) {
```

Comment:
zrobiłbym `isAvailabilityDeclarationAllowed` i tutaj odwrocil, bo gdzie bedzie trzeba na odwrot i bedzie niefajne podwojne not

ogolnie to widze ze to bylo juz wczesniej, ale mozna przy okazji zmienic

Reply:
W sumie też mi się nie podoba, chociaż to chyba ja sugerowałam, żeby to odwrócić oryginalnie, bo było używane tylko odwrotnie